### PR TITLE
refactor(strict): serialize primitives in strict mode if not serialize: false

### PR DESCRIPTION
**Pull request by ethanresnick**
Per #17

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -34,43 +34,34 @@ test('throws when passed non-object number using [strict: true]', ({ end, throws
...
-  end()
-})
+test('returns serialized, original value when passed non-object using [strict: false]', ({ end, is, doesNotThrow }) => {
+  const customSerialize = (v) => `Hello ${v}!`
+  const redactDefaultSerialize = fastRedact({ paths: ['a.b.c'], strict: false })
+  const redactCustomSerialize = fastRedact({
+    paths: ['a.b.c'],
+    strict: false,
+    serialize: customSerialize
+  })
```
Comment (mcollina) [inline test/index.js]: Can you please split this test into two, depending if there is a serialize or not?

**Comment by ethanresnick** [comment]
Ok, I've split up the tests. The tests that I removed, I removed because they no longer reflected the code's (intended) behavior.

File: readme.md [hunk 2]
```diff
@@ -157,8 +157,9 @@ console.log(o) // { a: 1, b: 2 }
 #### `strict` – `Boolean` - `[true]`
 The `strict` option, when set to `true`, will cause the redactor function to throw if instead 
 of an object it finds a primitive. When `strict` is set to `false`, the redactor function 
-will return the primitive value without being redacted and if the primitive is of type "string",
-the primitive value is returned as JSON (i.e. within double quotes and escaped per JSON rules).
+will treat the primitive value as having already been redacted, and return it serialized (with
+`JSON.stringify` or the user's custom `serialize` function), or as-is if the `serialize` option
+was set to false.
```
Comment (davidmarkclements) [inline readme.md]: I think we need to add a test for if serialize is `false`

**Comment by davidmarkclements** [comment]
sorry for the approval and then request change, something caught my eye after

File: readme.md (hunk 2 above)
Comment (ethanresnick) [reply readme.md]: Good catch. That case actually wasn't passing. I've added a test and fixed it.

**Comment by davidmarkclements** [comment]
thanks @ethanresnick great work! released as 1.5.0

**Comment by ethanresnick** [comment]
Thanks @davidmarkclements! Should this have been a sem-ver major change, though? As I mentioned in #17, it does change the output in some cases

**Comment by davidmarkclements** [comment]
technically yes... but with such a narrow edge case I think a minor is okay, since the practical impact of the change is likely zero

**Pull request merged**